static/js/services/db-sync.js

```diff
@@ -30,26 +30,33 @@ angular.module('inboxServices').factory('DBSync',
       return _.findWhere(errors, { status: 401 });
     };
 
-    var getOptions = function(direction, options) {
-      options = options || {};
-      _.defaults(options, {
+    var getOptions = function(direction) {
+      var options = {
         live: true,
         retry: true,
         timeout: false,
         heartbeat: 10000,
         back_off_function: backOffFunction
-      });
+      };
       if (direction === 'to') {
+        options.checkpoint = 'source';
+        options.filter = function(doc) {
+          // don't try to replicate read only docs back to the server
+          return READ_ONLY_TYPES.indexOf(doc.type) === -1 &&
+                 READ_ONLY_IDS.indexOf(doc._id) === -1 &&
+                 doc._id.indexOf(DDOC_PREFIX) !== 0;
+        };
         return $q.resolve(options);
       }
```

Comment:
does the `direction === 'from'` code start here?  To me this would be clearer in an `else` block.